test/unittests/skills/test_intent_service.py

```diff
@@ -68,5 +71,66 @@ def test_remove_context(self):
         self.assertEqual(len(self.context_manager.frame_stack), 0)
 
 
-if __name__ == '__main__':
-    unittest.main()
+def check_converse_request(message, skill_id):
+    return (message.msg_type == 'skill.converse.request' and
+            message.data['skill_id'] == skill_id)
+
+
+class ConversationTest(TestCase):
+    def setUp(self):
+        bus = mock.Mock()
+        self.intent_service = IntentService(bus)
+        self.intent_service.add_active_skill('atari_skill')
+        self.intent_service.add_active_skill('c64_skill')
+
+    def test_converse(self):
+        """Check that the _converse method reports if the utterance is handled.
+
+        Also check that the skill that handled the query is moved to the
+        top of the active skill list.
+        """
+        result = None
+
+        def runner(utterances, lang, message):
+            nonlocal result
+            result = self.intent_service._converse(utterances, lang, message)
+
+        hello = ['hello old friend']
+        utterance_msg = Message('recognizer_loop:utterance',
+                                data={'lang': 'en-US',
+                                      'utterances': hello})
+        t = Thread(target=runner, args=(hello, 'en-US', utterance_msg))
+        t.start()
+        time.sleep(0.5)
+        self.intent_service.handle_converse_response(
+            Message('converse.response', {'skill_id': 'c64_skill',
+                                          'result': False}))
+        time.sleep(0.5)
+        self.intent_service.handle_converse_response(
+            Message('converse.response', {'skill_id': 'atari_skill',
+                                          'result': True}))
+        t.join()
+        self.assertEqual(self.intent_service.active_skills[0][0],
```

Comment:
Consider assigning the value in self.intent_service.active_skills[0][0] to a variable to give context.  I for one have no idea what this is referring to.  This applies elsewhere in test cases as well.